### PR TITLE
[SPARK-2312] Logging Unhandled messages

### PR DESCRIPTION
**Pull request by isaias**
The  previous solution has changed based on https://github.com/apache/spark/pull/2048 discussions.

### CODE REVIEW

**Comment by andrewor14** [comment]
ok to test...

**Comment by andrewor14** [comment]
@isaias I think logging these messages are fine, but as @ScrapCodes indicated it will be fairly noisy for random dis/association events. I agree that having something logged is better than silently dropping unknown messages, but maybe we need to filter some of these out before we do that.

File: core/src/main/scala/org/apache/spark/util/ActorLogReceive.scala [hunk 1]
```diff
@@ -43,7 +43,13 @@ private[spark] trait ActorLogReceive {
 
     private val _receiveWithLogging = receiveWithLogging
 
-    override def isDefinedAt(o: Any): Boolean = _receiveWithLogging.isDefinedAt(o)
+    override def isDefinedAt(o: Any): Boolean = {
+      val handled = _receiveWithLogging.isDefinedAt(o)
+      if (!handled) {
+        log.warn(s"Received unexpected actor system event: $o")
```
Comment (srowen) [inline core/src/main/scala/org/apache/spark/util/ActorLogReceive.scala]: Yes, perhaps at least turn this down to `info`?

Comment (isaias) [inline core/src/main/scala/org/apache/spark/util/ActorLogReceive.scala]: @andrewor14 @srowen I think that down to info or debug is ok. What do you think? Can I change it? Wait for now?

**Comment by srowen** [comment]
@isaias can you make the change here? let's use debug logging.

**Comment by isaias** [comment]
@srowen done

**Comment by srowen** [comment]
Cool, unless anyone pops up to object, I'll merge tomorrow.

**Comment by andrewor14** [comment]
My only concern with logging at debug is that this will mask unexpected events in most cases. However, given that this is strictly more informative than before and we haven't had to rely on these warnings to debug the system, I think this is fine to merge as is.

**Comment by srowen** [comment]
Yeah, end users could turn on debug logging for this logger if needed. I could be turned up later to info by default if desired.

**Pull request closed without merging**